src/extensions/agent_modules/cloud_communication/eg_mqtt_broker_client/src/adu_enrollment_management.c

```diff
@@ -152,22 +107,65 @@ void OnMessage_enr_resp(
 
     if (!ParseEnrollmentMessagePayload(msg->payload, &isEnrolled, &scopeId))
     {
-        Log_Error("Failed parse of msg payload: %s", (char*)msg->payload);
+        Log_Error("Fail parse of msg payload: %s", (char*)msg->payload);
         goto done;
     }
 
     if (!Handle_Enrollment_Response(
         enrollmentData,
         isEnrolled,
         scopeId,
-        context))
+        retriableOperationContext))
     {
-        Log_Error("Failed handling enrollment response.");
+        Log_Error("Fail handling enrollment response.");
         goto done;
     }
 
 done:
     free(scopeId);
+    free(correlationData);
+}
+
+void OnPublish_enr_resp(struct mosquitto* mosq, void* obj, const mosquitto_property* props, int reason_code)
```

Comment:
>OnPublish_enr_resp [](http://example.com/codeflow?start=5&length=18)

this is almost a copy of OnPublish_ainfo_resp; some refactoring in order?

Reply:
Factored out into adu_mqtt_common.c now with this commit: [factor out puback handling to adu_mqtt_common](https://github.com/Azure/iot-hub-device-update/pull/584/commits/bd1513cbd9eadca3c66734cad1cfdf88fe227371)